Framework/src/QualitySender.cxx

```diff
@@ -0,0 +1,69 @@
+#include "QualityControl/QualitySender.h"
+#include <Framework/DataRefUtils.h>
+#include <Framework/InputRecordWalker.h>
+#include "QualityControl/QualitiesToFlagCollectionConverter.h"
+#include "QualityControl/QualityObject.h"
+
+namespace o2::quality_control::flag_sender
+{
+
+QualitySender::QualitySender(const std::string& grpcUri) : mGrpcUri{ grpcUri } {}
+
+void QualitySender::run(framework::ProcessingContext& context)
+{
+  for (auto const& ref : framework::InputRecordWalker(context.inputs())) {
+    try {
+      mQualityObjects.push_back(std::move(framework::DataRefUtils::as<core::QualityObject>(ref)));
+    } catch (...) {
+    }
+  }
+}
+
+auto merge(std::unique_ptr<QualityControlFlagCollection> collection, std::shared_ptr<core::QualityObject> qualityObject) -> std::unique_ptr<QualityControlFlagCollection>
+{
+  core::QualitiesToFlagCollectionConverter converter(std::move(collection), qualityObject->getPath());
+  converter(*qualityObject);
+  return converter.getResult();
+}
+
+auto collectionFromQualityObject(const core::QualityObject& qualityObject) -> std::unique_ptr<QualityControlFlagCollection>
+{
+  return std::make_unique<QualityControlFlagCollection>(
+    qualityObject.getName(),
+    qualityObject.getDetectorName(),
+    core::gFullValidityInterval, // we don't need exact run duration because each flag have it's own range.
+    qualityObject.getActivity().mId,
+    qualityObject.getActivity().mPeriodName,
+    qualityObject.getActivity().mPassName,
+    qualityObject.getActivity().mProvenance);
+}
+
+auto to_key(const core::QualityObject& qualityObject) -> std::string
+{
+  return qualityObject.getDetectorName() + "//" + qualityObject.getActivity().mPassName + "//" + qualityObject.getName();
+}
+
+using FlagsMap = std::unordered_map<std::string, std::unique_ptr<QualityControlFlagCollection>>;
+
+void send(const std::string grpcUri, const FlagsMap& flags)
+{
+  for (const auto& [_, flagCollection] : flags) {
+    for (const auto& flag : *flagCollection) {
+      //TODO: fill with bookkeeping api when ready
+    }
+  }
+}
+
+void QualitySender::stop()
```

Comment:
I would also do the same in case of an EndOfStream and ensure we do it just once. It's a matter of implementing also another callback, like this one: https://github.com/AliceO2Group/QualityControl/blob/cc46357fceebcf8f0e00289e8a96d06756122f00/Framework/src/TaskRunner.cxx#L320

EndOfStream is considered the right moment to wrap up knowing that no more data will come. `stop` can be the last resort, which in principle should always be called anyway, but I would rather not rely just on that.